crates/compression/src/eviction_policy.rs

```diff
@@ -0,0 +1,41 @@
+use std::collections::HashSet;
+
+use fuel_core_types::fuel_compression::RegistryKey;
+
+use crate::{
+    ports::EvictorDb,
+    tables::{
+        PerRegistryKeyspace,
+        RegistryKeyspace,
+    },
+};
+
+pub struct CacheEvictor {
+    /// Set of keys that must not be evicted
+    pub keep_keys: PerRegistryKeyspace<HashSet<RegistryKey>>,
+}
+
+impl CacheEvictor {
+    /// Get a key, evicting an old value if necessary
+    pub fn next_key<D>(
+        &mut self,
+        db: &mut D,
+        keyspace: RegistryKeyspace,
+    ) -> anyhow::Result<RegistryKey>
+    where
+        D: EvictorDb,
+    {
+        // Pick first key not in the set
+        // TODO: use a proper algo, maybe LRU?
+        let mut key = db.read_latest(keyspace)?;
```

Comment:
It seems you can cache the latest key during the first `next_key` and use local value instead doing it each time

Reply:
I'm not sure if optimizations like this make sense. The DB is supposed to be quite fast, and trading off code readibility for speed is not a thing I like to do without a good reason.

Reply:
Ok now done in eda9e77797